### PR TITLE
fix shufflebuild

### PR DESCRIPTION
**Pull request by badboynt1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/sql/colexec/shufflebuild/types.go [hunk 1]
```diff
@@ -99,21 +98,37 @@ func (shuffleBuild *ShuffleBuild) Release() {
 }
 
 func (shuffleBuild *ShuffleBuild) Reset(proc *process.Process, pipelineFailed bool, err error) {
-	shuffleBuild.Free(proc, pipelineFailed, err)
+	ctr := shuffleBuild.ctr
```
Comment (ouyuanning) [inline pkg/sql/colexec/shufflebuild/types.go:101]: 是要这样  ctr := &shuffleBuild.ctr ？

File: pkg/sql/colexec/shufflebuild/types.go [hunk 2]
```diff
@@ -99,21 +98,37 @@ func (shuffleBuild *ShuffleBuild) Release() {
...
+	ctr.state = ReceiveBatch
+	message.FinalizeRuntimeFilter(shuffleBuild.RuntimeFilterSpec, pipelineFailed, err, proc.GetMessageBoard())
+	message.FinalizeJoinMapMessage(proc.GetMessageBoard(), shuffleBuild.JoinMapTag, true, shuffleBuild.ShuffleIdx, pipelineFailed, err)
+	if ctr.batches != nil {
+		ctr.batches = ctr.batches[:0]
+	}
+	if ctr.buf != nil {
+		ctr.buf.CleanOnlyData()
+	}
+	ctr.intHashMap = nil
```
Comment (ouyuanning) [inline pkg/sql/colexec/shufflebuild/types.go:111]: reset时候感觉也要free掉他们吧

Comment (badboynt1) [reply pkg/sql/colexec/shufflebuild/types.go:111]: 目前没有啥区别，因为有shuffle算子的一定是ap，就一定不会走到reset上。  如果以后ap也支持prepare复用，那么这样写应该也可以

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/colexec/shufflebuild/types.go (hunk 2 above)
Comment (fengttt) [reply pkg/sql/colexec/shufflebuild/types.go:111]: I find that you ALWAYS have a model in your mind about an operator is AP/TP.   Seriously, WHY do you ever want to make such a distinction?

File: pkg/sql/colexec/shufflebuild/types.go [hunk 3]
```diff
@@ -99,21 +98,37 @@ func (shuffleBuild *ShuffleBuild) Release() {
 }
 
 func (shuffleBuild *ShuffleBuild) Reset(proc *process.Process, pipelineFailed bool, err error) {
-	shuffleBuild.Free(proc, pipelineFailed, err)
+	ctr := &shuffleBuild.ctr
+	ctr.state = ReceiveBatch
+	message.FinalizeRuntimeFilter(shuffleBuild.RuntimeFilterSpec, pipelineFailed, err, proc.GetMessageBoard())
+	message.FinalizeJoinMapMessage(proc.GetMessageBoard(), shuffleBuild.JoinMapTag, true, shuffleBuild.ShuffleIdx, pipelineFailed, err)
+	if ctr.batches != nil {
+		ctr.batches = ctr.batches[:0]
```
Comment (fengttt) [inline pkg/sql/colexec/shufflebuild/types.go:106]: Who owns all the batches stored here?

Comment (badboynt1) [reply pkg/sql/colexec/shufflebuild/types.go:106]: 发走以后这些batch就归join probe管理了。  这里把batches这个slice清空，代表的是build算子已经失去了对这些batch的管辖权。 至于是batches=nil  还是batches=[:0]  没什么区别

File: pkg/sql/colexec/shufflebuild/types.go (hunk 2 above)
Comment (badboynt1) [reply pkg/sql/colexec/shufflebuild/types.go:111]: 我是回答远宁的问题啊  这里可以区分aptp,也可以不区分aptp

**Pull request merged**